File: Palindrome/mine.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        reversed_number = 0
        temp = x
        if x < 0:
            return False
        elif x < 10:
            return True
        else:
            while temp != 0:
                last_digit = temp % 10
                temp = temp // 10
                reversed_number = (reversed_number * 10) + last_digit
        print(f"{reversed_number}")
        if x == reversed_number:
            return True
        return False

File: Palindrome/mine.java

public class Main {

    public static void main(String[] args) {
        int a = 121, b = 1, c = -121, d = 10;
        System.out.println(Solution.isPalindrome(a));
        System.out.println(Solution.isPalindrome(b));
        System.out.println(Solution.isPalindrome(c));
        System.out.println(Solution.isPalindrome(d));
    }
}

class Solution {

    public static boolean isPalindrome(int number) {
        if (number < 0 || (number % 10 == 0 && number != 0)) {
            return false;
        }
        int reversed = 0;
        while (number > reversed) {
            reversed = reversed * 10 + number % 10;
            number = number / 10;
        }
        return number == reversed || number == reversed / 10;
    }
}
